presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -576,13 +590,13 @@ public boolean equals(Object o)
 
         TypeSignature other = (TypeSignature) o;
 
-        return Objects.equals(this.base.toLowerCase(ENGLISH), other.base.toLowerCase(ENGLISH)) &&
+        return Objects.equals(this.base.toString().toLowerCase(ENGLISH), other.base.toString().toLowerCase(ENGLISH)) &&
```

Comment:
implement equals and hashCode on TypeSignatureBase instead of casting to string here?